src/security/SConscript

```diff
@@ -10,20 +20,23 @@ def scons():
     env.AppendUnique(LIBPATH=[Dir('.')])
     denv = env.Clone()
 
-    # Shared src between server and client
-    common_src = denv.SharedObject(['security.pb-c.c'])
+    # Populate the object cache
+    cache = {}
+    for src_file in SRC_FILES:
+        cache[src_file] = denv.SharedObject(src_file)
 
     ds_sec = daos_build.library(denv, 'security',
-                                ['srv.c', 'srv_acl.c', 'acl_api.c'])
+                                get_objects(cache, 'srv.c', 'srv_acl.c',
+                                            'acl_api.c'))
     denv.Install('$PREFIX/lib/daos_srv', ds_sec)
 
     # dc_security: Security Client
-    dc_security_tgts = denv.SharedObject(['cli_security.c',
-                                          'acl_api.c']) + common_src
+    dc_security_tgts = get_objects(cache, 'acl_api.c', 'cli_security.c',
+                                   'security.pb-c.c')
     Export('dc_security_tgts')
 
-    dc_sectest_tgts = denv.SharedObject(['cli_security.c', 'srv_acl.c'],
-                                        OBJPREFIX="s_") + common_src
```

Comment:
Still okay without the OBJPREFIX? We were building different copies to accommodate some build flags used by the unit tests.

Reply:
I don't see any difference in build flags between the copies.  They are all using the same construction environment (denv).

In general, OBJPREFIX is fine but should only be used where you need to build an object in multiple ways.

Reply:
We do import these targets into the unit test scons files, where they are compiled with different flags. If you're not getting any build failures, though, it must be working fine this way.

Reply:
Right.   The export is exporting already built object files.   What you do in unit tests won't case them to get recompiled though they would be relinked.   If you need them to be compiled again with different flags, you need to export the sources rather than the objects.